compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/WebSocketUpgradeServiceValidator.java

```diff
@@ -55,6 +59,14 @@
 
     void validate() {
         ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) ctx.node();
+        if (serviceDeclarationNode.members().isEmpty()) {
+            DiagnosticInfo diagnosticInfo = new DiagnosticInfo(
+                    PluginConstants.CompilationErrors.TEMPLATE_CODE_GENERATION_HINT.getErrorCode(),
+                    PluginConstants.CompilationErrors.TEMPLATE_CODE_GENERATION_HINT.getError(),
+                    DiagnosticSeverity.INTERNAL);
+            ctx.reportDiagnostic(DiagnosticFactory.createDiagnostic(diagnosticInfo, serviceDeclarationNode.location(),
+                    List.of(new NonCatProperty(serviceDeclarationNode))));
```

Comment:
Let's just pass the `serviceDeclarationNode.location()` to the diagnostic. No need to pass additional properties. (see my comment in code action. It's related)